File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    len_t = len(t)
    len_p = len(p)
    answer = 0
    ref = 0
    
    while ref <= (len_t - len_p):
        if int(t[ref: ref + len_p]) <= int(p):
            answer += 1
        ref += 1
    
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>

using namespace std;

int solution(string t, string p) {
    int answer = 0;
    long long target = stoll(p);

    for (int i = 0; i < t.length() - p.length() + 1; i++) {
        if (stoll(t.substr(i, p.length())) <= target) answer++;
    }
    
    return answer;
}